src/Sylius/Bundle/ThemeBundle/Asset/Installer/OutputAwareAssetsInstaller.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ThemeBundle\Asset\Installer;
+
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class OutputAwareAssetsInstaller extends AssetsInstaller implements OutputAwareInterface
+{
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setOutput(OutputInterface $output)
+    {
+        $this->output = $output;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installAssets($targetDir, $symlinkMask)
+    {
+        if ($this->hasOutput()) {
+            $this->output->writeln($this->provideExpectationComment($symlinkMask));
+        }
+
+        return parent::installAssets($targetDir, $symlinkMask);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installBundleAssets(BundleInterface $bundle, $targetDir, $symlinkMask)
+    {
+        $sources = $this->findAssetsPaths($bundle);
+
+        if (empty($sources)) {
+            return $symlinkMask;
+        }
+
+        if ($this->hasOutput()) {
+            $this->output->writeln(sprintf('Installing assets for <comment>%s</comment>', $bundle->getNamespace(), $targetDir));
+        }
+
+        $effectiveSymlinkMask = parent::installBundleAssets($bundle, $targetDir, $symlinkMask);
+
+        if ($this->hasOutput()) {
+            $this->output->writeln($this->provideResultComment($symlinkMask, $effectiveSymlinkMask));
+        }
+
+        return $effectiveSymlinkMask;
+    }
+
+    /**
+     * @return bool
+     */
+    private function hasOutput()
+    {
+        return null !== $this->output;
+    }
+
+    /**
+     * @param int $symlinkMask
+     * @param int $effectiveSymlinkMask
+     *
+     * @return string
+     */
+    private function provideResultComment($symlinkMask, $effectiveSymlinkMask)
+    {
+        if ($effectiveSymlinkMask === $symlinkMask) {
+            switch ($symlinkMask) {
+                case AssetsInstallerInterface::HARD_COPY:
+                    return 'The assets were copied.';
+                case AssetsInstallerInterface::SYMLINK:
+                    return 'The assets were installed using symbolic links.';
+                case AssetsInstallerInterface::RELATIVE_SYMLINK:
+                    return 'The assets were installed using relative symbolic links.';
```

Comment:
I would rather use properly written array with ``$symlinkMask`` as a key and message as a value, but maybe it's only me :smile: